lib/readdir.js

```diff
@@ -0,0 +1,120 @@
+const syscall = require('./syscall');
+const bindings = require('bindings')('bindings');
+
+
+const O_DIRECTORY = 00200000;
+const O_RDONLY = 00000000;
+const openOptions = O_DIRECTORY | O_RDONLY;
+const bufferSize = 1024;
+const notStringOrBufferError = TypeError(
+	"TypeError: Path must be a string or a Buffer."
+);
+const invalidEncodingError = new Error(
+	'EINVAL: Invalid character encoding for filename'
+);
+invalidEncodingError.code = 'EINVAL';
+invalidEncodingError.errno = 22;
+
+
+function decodeArray(arr, encoding) {
+	// Decodes each Buffer in arr using the encoding argument (in place).
+	let s;
+	if (encoding !== 'buffer') {
+		for (let i=0; i<arr.length; i++) {
+			s = arr[i].toString(encoding);
+			if (s.length === 0) {
+				throw invalidEncodingError;
```

Comment:
it happens here https://github.com/nodejs/node/blob/master/src/node_file.cc#L964-L969